builtin/credential/aws/path_role_test.go

```diff
@@ -986,6 +986,65 @@ func TestAwsVersion(t *testing.T) {
 	}
 }
 
+// This test was used to reproduce https://github.com/hashicorp/vault/issues/7418
+// and verify its fix.
+func TestRoleResolutionWithSTSEndpointConfigured(t *testing.T) {
+	t.Skip("skipping test because it hits real endpoints")
+
+	config := logical.TestBackendConfig()
+	storage := &logical.InmemStorage{}
+	config.StorageView = storage
+
+	b, err := Backend(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = b.Setup(context.Background(), config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// configure the client with an sts endpoint that should be used in creating the role
+	data := map[string]interface{}{
+		"sts_endpoint": "https://sts.eu-west-1.amazonaws.com",
+		// Note - if you comment this out, you can reproduce the error shown
+		// in the linked GH issue above. This essentially reproduces the problem
+		// we had when we didn't have an sts_region field.
+		"sts_region": "eu-west-1",
+	}
+	resp, err := b.HandleRequest(context.Background(), &logical.Request{
+		Operation: logical.CreateOperation,
+		Path:      "config/client",
+		Data:      data,
+		Storage:   storage,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if resp != nil && resp.IsError() {
+		t.Fatalf("failed to create the role entry; resp: %#v", resp)
+	}
+
+	data = map[string]interface{}{
+		"auth_type":               iamAuthType,
+		"bound_iam_principal_arn": "arn:aws:iam::123456789012:assumed-role/MyRole/foo",
```

Comment:
If you want to test this for real, this will also need to be updated, right? Might be worth leaving a comment to that effect, both here and in the `t.Skip` line to help people in the future know what they need to do if they want to run this test "for real" and not skip it.

Also, this gets into some AWS arcana, but in general: `iam` means `role` and `sts` means `assumed-role`. So, this would likely be better expressed as `arn:aws:iam::123456789012:role/MyRole`.